spec.bs

```diff
@@ -1146,10 +1146,38 @@ To <dfn>validate and convert auction ad config</dfn> given an {{AuctionAdConfig}
 1. If |seller| is failure, then return failure.
 1. Set |auctionConfig|'s [=auction config/seller=] to |seller|.
 1. If |config|["{{AuctionAdConfig/serverResponse}}"] [=map/exists=]:
-  1. If |config|["{{AuctionAdConfig/requestId}}"] does not [=map/exist=], then [=exception/throw=]
-    a {{TypeError}}.
-  1. Set |auctionConfig|'s [=auction config/server response=] to |config|["{{AuctionAdConfig/serverResponse}}"].
-  1. Set |auctionConfig|'s [=auction config/server response id=] to |config|["{{AuctionAdConfig/requestId}}"].
+  1. If |config|["{{AuctionAdConfig/requestId}}"] does not [=map/exist=], then [=exception/throw=] a
+    {{TypeError}}.
+  1. Let |resolvedAndTypeChecked| be the promise representing performing the following steps
+    [=upon fulfillment=] of |config|["{{AuctionAdConfig/serverResponse}}"] with |result|:
+    1. Set |auctionConfig|'s [=auction config/server response=] to the [=ArrayBuffer/transferred=]
+      |config|["{{AuctionAdConfig/serverResponse}}"].
+    1. Set |auctionConfig|'s [=auction config/server response id=] to
+      |config|["{{AuctionAdConfig/requestId}}"].
+    1. If no exception was [=exception/thrown=] in the previous step, then decrement
+      |auctionConfig|'s [=auction config/pending promise count=].

```

Comment:
I think you don't want to touch the pending promise count here, since you're using a separate kind of wait for this; and besides, you also never incremented it in the first place. This step is also confusing WRT to whether rejection runs or not.

Reply:
ah yes. Forgot to remove it.